src/Components/Facility/Consultations/PressureSoreDiagrams.tsx

```diff
@@ -45,6 +42,11 @@ export const PressureSoreDiagrams = (props: any) => {
     [consultationId, dispatch, offset]
   );
 
+  useEffect(() => {
+    if (Object.keys(results).length > 0)
+      setSelectedDateData(results, Object.keys(results?.["results"])[0]);
+  }, [results]);
+
```

Comment:
I didn't understood why there is separate `results` variable if `selectedDate` and `results`'s data are to be sync by this code. Also there is only one call of `setResult`. Can you make it as no need of `results` ?

Also there is `dates` variable which sets on same condition(`Object.keys.length > 0`) below this part of code

Reply:
`dates` variable is used for dropdown. `results` state is updated every time we change date in dropdown

Reply:
Ohk got it